examples/bigquery_user_info_updater_tool/bigquery_user_info_updater/tests/test_query_creator.py

```diff
@@ -0,0 +1,208 @@
+
+         Args:
+             project_id(str): ID of the project that holds the test BQ tables.
+
+         Returns:
+             True if test passes, else False.
+         """
+        abs_path = os.path.abspath(os.path.dirname(__file__))
+        schema_path = os.path.join(
+            abs_path,
+            'test_schemas/test_nested_schema.json'
+        )
+        test_query_creator = query_creator.QueryCreator(
+            schema_path=schema_path,
+            user_id_field_name=self.user_id_field_name,
+            ingest_timestamp_field_name=self.ingest_timestamp_field_name,
+            project_id=project_id,
+            dataset_id=self.dataset_id,
+            updates_table_id=self.user_info_updates_id,
+            temp_updates_table_id=self.temp_user_info_updates_id,
+            final_table_id=self.user_info_final_id
+        )
+
+        gather_updates_query = test_query_creator.create_gather_updates_query()
+        abs_path = os.path.abspath(os.path.dirname(__file__))
+        test_gather_updates_query_path = os.path.join(
+            abs_path,
+            'test_queries/test_nested_gather_updates_query.txt')
+        with open(test_gather_updates_query_path, 'r') as input_file:
+            expected_gather_updates_query = input_file.read().format(
+                project_id,
+                '{0:s}'
+            )
+        assert gather_updates_query == expected_gather_updates_query
+
+    def test_create_merge_query(self, project_id):
+        """Tests QueryCreator's ability to create a query to merge updates.
+
+        Args:
+            project_id(str): ID of the project that holds the test BQ tables.
+
+        Returns:
+            True if test passes, else False.
+        """
+        abs_path = os.path.abspath(os.path.dirname(__file__))
+        schema_path = os.path.join(
+            abs_path,
+            'test_schemas/test_schema.json'
+        )
+
+        test_query_creator = query_creator.QueryCreator(
+            schema_path=schema_path,
+            user_id_field_name=self.user_id_field_name,
+            ingest_timestamp_field_name=self.ingest_timestamp_field_name,
+            project_id=project_id,
+            dataset_id=self.dataset_id,
+            updates_table_id=self.user_info_updates_id,
+            temp_updates_table_id=self.temp_user_info_updates_id,
+            final_table_id=self.user_info_final_id
+        )
+
+        merge_updates_query = test_query_creator.create_merge_query()
+        self._write_query('merge.txt', merge_updates_query)
+        abs_path = os.path.abspath(os.path.dirname(__file__))
+        test_merge_updates_query_path = os.path.join(
+            abs_path,
+            'test_queries/test_merge_updates_query.txt')
+        with open(test_merge_updates_query_path, 'r') as input_file:
+            expected_merge_updates_query = input_file.read().format(
+                project_id
+            )
+        assert merge_updates_query == expected_merge_updates_query
```

Comment:
jaketf on Aug 27, 2019 Member
use self.assertEqual

 @annarudy
annarudy on Oct 15, 2019 Author Member
self.assertEquals requires the test class to inherit from unittest.TestCase. When those modifications are made, the project_id param can't be passed in when the tests are run, resulting in errors.